shimmingtoolbox/coils/spher_harm_basis.py

```diff
@@ -214,15 +243,25 @@ def _reorder_to_ge(spher_harm):
         numpy.ndarray: Coefficients ordered following GE convention
     """
 
-    if spher_harm.shape[-1] != 8:
-        raise RuntimeError("Input arrays should have 4th dimension's shape equal to 8")
+    if orders == (1, 2):
+        if spher_harm.shape[-1] != 8:
+            raise RuntimeError("Input arrays should have 4th dimension's shape equal to 8")
+        reordered = spher_harm[..., [2, 0, 1, 3, 4, 6, 7, 5]]
+
+    if orders == (2,):
+        if spher_harm.shape[-1] != 5:
+            raise RuntimeError("Input arrays should have 4th dimension's shape equal to 5")
+        reordered = spher_harm[..., [0, 1, 3, 4, 2]]
 
-    reordered = spher_harm[..., [2, 0, 1, 3, 4, 6, 7, 5]]
+    if orders == (1,):
+        if spher_harm.shape[-1] != 3:
+            raise RuntimeError(f"Input arrays should have 4th dimension's shape equal to 3 {spher_harm.shape}")
+        reordered = spher_harm[..., [2, 0, 1]]
 
     return reordered
 
 
-def scaled_spher_harm(x, y, z, orders=(1, 2), shim_cs='ras'):
+def scaled_spher_harm(x, y, z, all_orders, shim_cs='ras'):
```

Comment:
If the input spherical harmonic order is 2, the scaling factor that gets applied is wrong. (scaling_factors always has 8 coefficients)